### PR TITLE
Add packages for dw-ingestion-dataset-generation

### PR DESCRIPTION
**Pull request by tea-short**

### CODE REVIEW

**Review by ChrisScottThomas — COMMENTED** [review]
I'm not sure you should do this here.  This is the base of all other AMI's and there's no telling what knock on effect this will have on those AMIs.  Perhaps I am being a bit overly cautious, but I think that if you want to install new packages for a single purpose, then that is not a base requirement of all AMIs.

**Comment by benjaminsherwood** [comment]
Have to disagree, think this has to be done here - either that or @jimbarker needs to agree to lowering the security protection. We currently block pip install access via the proxy. Ultimately, we should be fine and happy having `python3` in our base AMI - it's not changing the default Python from 2.7, and Linux can exist quite happily with multiple versions of Python.

If we are happy allowing the proxy access to pip install, then we should only install `python3` and move the `pip` installs to the `userdata`, but otherwise it's going to be stuck here. `jinja2` and `pyyaml` are commonly/widely used packages which will find use in other scripts/places in our deployments, I would imagine. As they're annoyingly required here, we may need to accept having them installed.

@mattburgess @tea-short @ChrisScottThomas

**Comment by tea-short** [comment]
> I'm not sure you should do this here. This is the base of all other AMI's and there's no telling what knock on effect this will have on those AMIs. Perhaps I am being a bit overly cautious, but I think that if you want to install new packages for a single purpose, then that is not a base requirement of all AMIs.



> Have to disagree, think this has to be done here - either that or @jimbarker needs to agree to lowering the security protection. We currently block pip install access via the proxy. Ultimately, we should be fine and happy having `python3` in our base AMI - it's not changing the default Python from 2.7, and Linux can exist quite happily with multiple versions of Python.
> 
> If we are happy allowing the proxy access to pip install, then we should only install `python3` and move the `pip` installs to the `userdata`, but otherwise it's going to be stuck here. `jinja2` and `pyyaml` are commonly/widely used packages which will find use in other scripts/places in our deployments, I would imagine. As they're annoyingly required here, we may need to accept having them installed.
> 
> @mattburgess @tea-short @ChrisScottThomas

Agree with the above. Python3 should be a part of the base image and jinja2 and pyyaml as well.

**Comment by mattburgess** [comment]
FWIW I can absolutely see both sides of the argument here. But, knowing that Python3 will be installed completely separately from the system python, I'm actually quite comfortable with this change.

**Review by mattburgess — APPROVED** [review]

**Comment by mattburgess** [comment]
It also has the advantage that we can start to be good citizens and write our scripts using Python3, meaning they won't all blow up when AL3 comes out and drops Python2.7 support completely.

File: dw-al2-base-ami/dw-base-ami-install.sh [hunk 1]
```diff
@@ -40,6 +40,10 @@ sed -i.bak -e 's/repo_upgrade: security/repo_upgrade: none/' \
 
 yum install -y python-pip gcc yum-plugin-remove-with-leaves sudo
```
Comment (ChrisScottThomas) [inline dw-al2-base-ami/dw-base-ami-install.sh:41]: this still installs `python-pip` from `python2`.  Is that still needed?

Comment (tea-short) [reply dw-al2-base-ami/dw-base-ami-install.sh:41]: Both Python 2 and Python3 and pip2 and pip3 are needed.

**Comment by ChrisScottThomas** [comment]
> FWIW I can absolutely see both sides of the argument here. But, knowing that Python3 will be installed completely separately from the system python, I'm actually quite comfortable with this change.

I'm more fine, with more outside support.  The recent changes and issues has me edgey.

File: dw-al2-base-ami/dw-base-ami-install.sh (hunk 1 above)
Comment (ChrisScottThomas) [reply dw-al2-base-ami/dw-base-ami-install.sh:41]:
```suggestion
yum install -y python-pip gcc
```

OK, then I suggest the above as:

```
    amazon-ebs: Package yum-plugin-remove-with-leaves-1.1.31-46.amzn2.0.1.noarch already installed and latest version
    amazon-ebs: Package sudo-1.8.23-4.amzn2.2.x86_64 already installed and latest version
```
in the PR log.

**Comment by jimbarker** [comment]
If we're going to bake in the old version of python anyway, then should bake in the new version too. Prevents us from getting into bad habits and using v2.7 "just because it's there". As I understand it, python, pyyaml and jinja2 are pretty much ubiquitous requirements for our AMIs so understand them being in our base AMI. If the situation changes we should separate them out into application-specific AMIs for the AMIs which use them. However, happy with the current approach given the current situation.

File: dw-al2-base-ami/dw-base-ami-install.sh (hunk 1 above)
Comment (benjaminsherwood) [reply dw-al2-base-ami/dw-base-ami-install.sh:41]: If we're blocking access in the proxy, then I'd imagine not.

**Review by jimbarker — APPROVED** [review]

**Pull request merged**